syntax = "proto3";

package primitiv.messages;

message Shape {
  repeated uint32 dims = 1;
  uint32 batch = 2;
}

message Tensor {
  Shape shape = 1;
  repeated float data = 2;
}

message Parameter {
  string name = 1;
  Tensor value = 2;
  map<string, Tensor> stats = 3;
}

message TrainerConfigs {
  string name = 1;
  map<string, uint32> uint_configs = 2;
  map<string, float> float_configs = 3;
}
